core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingRoutesEndpoint.java

```
@@ -102,16 +106,43 @@ public static Pathfinding.Result<SignalingRoute> runPathfinding(
         }
 
+        Pathfinding.Result<SignalingRoute> pathfinding = computePaths(infra, waypoints, loadingGaugeConstraints,
+                electrificationConstraints, remainingDistanceEstimator);
         // Compute the paths from the entry waypoint to the exit waypoint
-        return new Pathfinding<>(new GraphAdapter<>(infra.getSignalingRouteGraph()))
+        return pathfinding;
+    }
+
+    private static Pathfinding.Result<SignalingRoute> computePaths(SignalingInfra infra,
+             ArrayList<Collection<Pathfinding.EdgeLocation<SignalingRoute>>> waypoints,
+             LoadingGaugeConstraints loadingGaugeConstraints,
+             ElectrificationConstraints electrificationConstraints,
+             RemainingDistanceEstimator remainingDistanceEstimator) {
+
+        var path = new Pathfinding<>(new GraphAdapter<>(infra.getSignalingRouteGraph()))
                 .setEdgeToLength(route -> route.getInfraRoute().getLength())
-                .addBlockedRangeOnEdges(loadingGaugeConstraints)
+                .setRemainingDistanceEstimator(remainingDistanceEstimator);
+        var pathfinding = path.addBlockedRangeOnEdges(loadingGaugeConstraints)
                 .addBlockedRangeOnEdges(electrificationConstraints)
-                .setRemainingDistanceEstimator(remainingDistanceEstimator)
                 .runPathfinding(waypoints);
+        // handling errors
+        if (pathfinding == null) {
+            var pathError = path.addBlockedRangeOnEdges(loadingGaugeConstraints).runPathfinding(waypoints);
+            if (pathError != null) {
+                throw new NoPathFoundError("No path could be found due to Gauge");
+            } else {
+                pathError = path.addBlockedRangeOnEdges(electrificationConstraints).runPathfinding(waypoints);
+                if (pathError != null) {
+                    throw new NoPathFoundError("No path could be found due to Electrification");
+                }
+            }
+            pathfinding = null;
```

`if (pathfinding ==null) pathfinding = null;` ? ;)

Anyway, isn't sometime returning null when not finding a path and sometime throwing an exception a little weird? Should we maybe throw an exception in the case there is not path at all too?